vissl/engines/slurm_evaluator.py

```diff
@@ -0,0 +1,506 @@
+        config_files.insert(1, f"config.SLURM.LOG_FOLDER='{log_dir}'")
+        config_files.insert(1, f"config.CHECKPOINT.DIR='{checkpoint_dir}'")
+        config_files.insert(1, f"hydra.run.dir='{ log_dir }'")
+        config_files.insert(1, "hydra.verbose=true")
+
+        evaluation_configs[training_checkpoint].append(benchmark_result)
+
+    def _evaluation_log_dir(self, evaluation_directory, evaluation_name):
+        """
+        Directory to put logs for an evaluation job.
+        """
+        evaluation_dir = self.evaluation_dir()
+        return os.path.join(evaluation_dir, evaluation_directory, evaluation_name)
+
+    def _evaluation_checkpoint_dir(self, model_final_checkpoint, evaluation_name):
+        """
+        Directory to put checkpoints in for an evaluation job.
+        """
+        return os.path.join(
+            self._evaluation_log_dir(model_final_checkpoint, evaluation_name),
+            "checkpoints",
+        )
+
+    def _generate_config(self, config):
+        """
+        Generate AttrDict config from a config YAML file and overrides.
+        """
+        with initialize_config_module(config_module="vissl.config"):
+            config = compose("defaults", overrides=config)
+
+        return convert_to_attrdict(config)
+
+class SlurmEvaluatorJob:
+    def __init__(self, slurm_evaluator: SlurmEvaluator):
+        self.slurm_evaluator = slurm_evaluator
+
+    def __call__(self):
+        self.slurm_evaluator.evaluate()
+
+    def checkpoint(self):
+        self.slurm_evaluator.save_evaluation_results()
+        self.slurm_evaluator.autoload_slurm_evaluator_checkpoint = True
+
+        trainer = SlurmEvaluatorJob(slurm_evaluator=self.slurm_evaluator)
+        return submitit.helpers.DelayedSubmission(trainer)
+
+def launch_slurm_evaluator(config_file):
+    assert PathManager.exists(config_file), "Slurm evaluator config file must exist."
+
+    config = load_file(config_file)
+    slurm_evaluator = SlurmEvaluator(config=config)
+    slurm_evaluator_job = SlurmEvaluatorJob(slurm_evaluator=slurm_evaluator)
+    executor = submitit.AutoExecutor(folder=slurm_evaluator.evaluation_dir())
+
+    override_slurm_options = config.get("slurm_options", {})
+    slurm_options = { **_DEFAULT_SLURM_OPTIONS,  **override_slurm_options }
+    slurm_options = AttrDict(slurm_options)
+
+    executor.update_parameters(
+        name=slurm_options.NAME,
+        slurm_comment=slurm_options.COMMENT,
+        slurm_partition=slurm_options.PARTITION,
+        slurm_constraint=slurm_options.CONSTRAINT,
+        timeout_min=slurm_options.TIMEOUT_MIN,
+        nodes=1,
+        cpus_per_task=slurm_options.CPUS_PER_TASK,
+        tasks_per_node=1,
+        mem_gb=slurm_options.MEM_GB,
+        slurm_additional_parameters=slurm_options.ADDITIONAL_PARAMETERS,
+    )
+
+    job = executor.submit(slurm_evaluator_job)
+    print(f"SUBMITTED EVALUATION JOB: {job.job_id}")
+
+    return job
+
+if __name__ == "__main__":
```

Comment:
The idea would be to create a new script under `extra_script` or `tool` folder (not sure which is the best) and move the `__main__` in this part.